### PR TITLE
Pass hasUserLocationConsent to DataCollector

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Add javadocs for `hasUserLocationConsent` param
 - Pass `hasUserLocationConsent` to `DataCollector.collectDeviceData()`

 ### Checklist

 - [ ] Added a changelog entry

-

### CODE REVIEW

File: Drop-In/src/main/java/com/braintreepayments/api/DropInInternalClient.java [hunk 1]
```diff
@@ -148,10 +148,8 @@ void shouldRequestThreeDSecureVerification(PaymentMethodNonce paymentMethodNonce
     void tokenizePayPalRequest(FragmentActivity activity, PayPalFlowStartedCallback callback) {
         PayPalRequest paypalRequest = dropInRequest.getPayPalRequest();
         if (paypalRequest == null) {
-            paypalRequest = new PayPalVaultRequest();
+            paypalRequest = new PayPalVaultRequest(false);
         }
-        // TODO: set hasUserLocationConsent on paypalRequest
-        dropInRequest.hasUserLocationConsent();
```
Comment (tdchow) [inline Drop-In/src/main/java/com/braintreepayments/api/DropInInternalClient.java:154]: `hasUserLocationConsent` is set on the `PayPalRequest` directly. No need to set it here.

Comment (sarahkoop) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInInternalClient.java:154]: If the merchant has set `hasUserLocationConsent` on their `dropInRequest` already - can we pass that value in when creating the default `PayPalVaultRequest`?

Comment (tdchow) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInInternalClient.java:154]: yep, great suggestion! i'll update this.

unrelated question - if this function gets called with a null `PayPalRequest` and a default one is created, are there scenarios where `payPalClient.tokenizePayPalAccount()` will ever succeed? are there required fields on the request?

Comment (sarahkoop) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInInternalClient.java:154]: Before this change, there were no required fields for a `PayPalVaultRequest` (checkout request requires amount). So if a merchant doesn't pass a custom `PayPalRequest` to their `DropInRequest`, drop-ins default behavior is the PayPal billing agreement/vault flow - it should succeed when we create the default vault request.

**Comment by sarahkoop** [comment]
Is there going to be a separate PR to pass this flag to all other instances of collectDeviceData throughout drop-in?

**Comment by tdchow** [comment]
> Is there going to be a separate PR to pass this flag to all other instances of collectDeviceData throughout drop-in?

I was intending this to be the final PR for drop-in. I wasn't able to find any other instances where we'd need to pass the flag when searching for "collectDeviceData" in the repo. Are there other callsites that I'm currently missing?

**Review by sarahkoop — APPROVED** [review]

**Review by saperi22 — APPROVED** [review]

**Pull request merged**